### PR TITLE
Add devcontainer to project

### LINKED ISSUE #1657: Add VSCode devcontainer configuration to project
## 1. General summary of the issue

Adding a recommended, optional devcontainer-configuration to the project might help current and future contributors to get started faster using VSCode while keeping their systems clean.

See https://code.visualstudio.com/docs/remote/containers and/or https://code.visualstudio.com/docs/remote/containers-tutorial

## 3. Expected Behavior

Open this project in VSCode on a computer with docker and automatically get a suitable dev environment including the right tools, SDKs and extensions to contribute to Pester. 

## 4.Current Behavior

Users have to install .NET Core ++ locally and manually before being able to build and contribute to the project.

## 5. Possible Solution

Add a devcontainer-configuration to the project which includes PowerShell, .NET Core, VS Code Extensions (Powershell + C#) ++.

Could use the [dotnetcore ](https://github.com/microsoft/vscode-dev-containers/tree/master/containers/dotnetcore/.devcontainer) sample devcontainer as a starting point.

## 6. Context

VSCode + [Remote - Container](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers) extension + Docker allows a developer to work on a project inside a specialized development container to keep a consistent environment for all contributors. The container image can include necessary build & test tools + useful extensions for VSCode itself.

Considering the project's build process now includes both PowerShell and C# using .NET Core, it might be time introduce this as it keeps a user's system clean while also potentially improving compilation/test performance.

### PR DESCRIPTION
**Pull request by fflaten**
## 1. General summary of the pull request

This PR adds devcontainer-configuration to the project. The devcontainer provides an isolated docker container to develop, build and test Pester. Using VSCode + "Remote - Container" extension, this environment is automatically loaded and seamless to the user. This helps contributors get up and running quickly with the required build tools like .NET Core 3.1 SDK. See linked issue for more information.

This first edition of the devcontainer is a lightly customized version of the [dotnetcore ](https://github.com/microsoft/vscode-dev-containers/tree/master/containers/dotnetcore/.devcontainer) sample container provided by Microsoft as it already includes .NET Core SDK 3.1 and PowerShell 7.0.3. The PR also adds a nuget-reference to  [Microsoft.NETFramework.ReferenceAssemblies](https://www.nuget.org/packages/Microsoft.NETFramework.ReferenceAssemblies/) to enable unix systems to build against .Net Framework 4.5.2

Fix #1657

### CODE REVIEW

File: src/csharp/Pester/Pester.csproj [hunk 1]
```diff
@@ -1,14 +1,19 @@
...
   </PropertyGroup>
-  
+
   <ItemGroup Condition="'$(TargetFramework)' == 'netstandard2.0'">
     <PackageReference Include="System.Management.Automation" Version="6.1.0" />
   </ItemGroup>
 
   <ItemGroup Condition="'$(TargetFramework)' == 'net452'">
     <PackageReference Include="Microsoft.PowerShell.3.ReferenceAssemblies" Version="1.0.0" />
+    <PackageReference Include="Microsoft.NETFramework.ReferenceAssemblies" Version="1.0.0">
```
[the remark this answers is not in the document]
Comment (nohwnd) [reply src/csharp/Pester/Pester.csproj:15]: I thought I already did that on v5.0, if I did not, then it's okay to remove it. Ideally in a separate PR.

[the remark this answers is not in the document]
Comment (fflaten) [reply src/csharp/Pester/Pester.csproj:15]: Removed in #1663 .

**Comment by nohwnd** [comment]
It's in :)

**Pull request merged**